chalice/awsclient.py

```diff
@@ -121,15 +132,55 @@ def update_function(self, function_name, zip_contents,
             # We're going to need this moving forward anyways,
             # more config options besides env vars will be added.
             'Environment': {'Variables': environment_variables},
-        }
+        }  # type: Dict[str, Any]
         if runtime is not None:
             kwargs['Runtime'] = runtime
+        if timeout is not None:
+            kwargs['Timeout'] = timeout
+        if memory_size is not None:
+            kwargs['MemorySize'] = memory_size
         lambda_client.update_function_configuration(**kwargs)
         if tags is not None:
-            lambda_client.tag_resource(Resource=return_value['FunctionArn'],
-                                       Tags=tags)
+            function_arn = return_value['FunctionArn']
+            remote_tags = lambda_client.list_tags(
+                Resource=function_arn)['Tags']
+
+            # Remove any tags that exist remotely but are not included
```

Comment:
Seems like this is just basic set/dict manipulation.  Can't this whole thing be simplified to something like:

```python
tags_to_remove = set(tags_requested) - set(remote_tags)
tags_to_add = {k: v for k, v in tags_requested.items()
              if k not in remote_tags or v != remote_tags[k]}
self._set_function_tags(tags_to_remove, tags_to_add)
```

I find this easier to read and think it's a simpler as well.